ipa-core/src/report.rs

```diff
@@ -413,6 +434,186 @@ where
     }
 }
 
+/// A binary report as submitted by a report collector, containing encrypted `OprfReport`
+/// An `EncryptedOprfReport` consists of:
+///     `ct_mk`: Enc(`match_key`)
+///     `ct_btt`: Enc(`breakdown_key`, `trigger_value`, `timestamp`)
+///     associated data of `ct_mk`: `key_id`, `epoch`, `event_type`, `site_domain`,
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct EncryptedOprfReport<BK, TV, TS, B>
+where
+    B: Deref<Target = [u8]>,
+    BK: SharedValue,
+    TV: SharedValue,
+    TS: SharedValue,
+{
+    data: B,
+    phantom_data: PhantomData<(BK, TV, TS)>,
+}
+
+/// follows the outline of the implementation of `EncryptedReport`
+impl<B, BK, TV, TS> EncryptedOprfReport<BK, TV, TS, B>
```

Comment:
We eventually want at least BK, and maybe also TV and TS to be runtime parameters, right? Are we tracking that work somewhere? (I think it will be a fair amount of work -- certainly not something that makes sense to do in this PR.)

Reply:
Would we compile all possible choices of BA types for them? I am not aware that we are tracking this somewhere.

Reply:
yes, likely we would use a macro for it

Reply:
https://github.com/private-attribution/ipa/issues/953